### PR TITLE
 Add v3 Transaction fields 

### PR DESCRIPTION
**Pull request by omerfirmak**

### CODE REVIEW

File: adapters/sn2core/sn2core.go [hunk 1]
```diff
@@ -157,15 +157,32 @@ func AdaptTransaction(transaction *starknet.Transaction) (core.Transaction, erro
...
+		NonceDAMode:           core.DataAvailabilityMode(t.NonceDAMode),
+		FeeDAMode:             core.DataAvailabilityMode(t.FeeDAMode),
+	}
+}
+
+func adaptResourceBounds(rb map[starknet.Resource]starknet.ResourceBounds) map[core.Resource]core.ResourceBounds {
+	coreBounds := make(map[core.Resource]core.ResourceBounds, len(rb))
+	for resource, bounds := range rb {
+		coreBounds[core.Resource(resource)] = core.ResourceBounds{
+			MaxAmount:       bounds.MaxAmount.Uint64(),
```
Comment (omerfirmak) [inline adapters/sn2core/sn2core.go:181]: is it safe to assume `bounds.MaxAmount` fits in a uint64?

File: core/transaction.go [hunk 2]
```diff
@@ -21,6 +21,47 @@ import (
...
+const (
+	WEI FeeUnit = iota
+	STRK
+)
+
+// From the RPC spec: The max amount and max price per unit of gas used in this transaction.
+type ResourceBounds struct {
+	MaxAmount uint64
+	// MaxPricePerUnit is technically a uint128
+	MaxPricePerUnit *felt.Felt
```
Comment (joshklop) [inline core/transaction.go:62]: Something I was wondering while implementing: Should we have a `uint128` type? Might save a little disk space.

**Review by joshklop — APPROVED** [review]

File: adapters/sn2core/sn2core.go (hunk 1 above)
Comment (joshklop) [reply adapters/sn2core/sn2core.go:181]: It's what the spec says https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-8.md

File: core/transaction.go (hunk 2 above)
Comment (omerfirmak) [reply core/transaction.go:62]: Yeah, could be worthwhile to explore later on.

File: adapters/sn2core/sn2core.go (hunk 1 above)
Comment (joshklop) [reply adapters/sn2core/sn2core.go:181]: We will have to do db migrations if we wait. Might be better to delay the release by 1-2 days and just implement them the first time imho.

Comment (omerfirmak) [reply adapters/sn2core/sn2core.go:181]: > We will have to do db migrations if we wait
Not really, felts are stored as big-endian byte arrays. We can keep using the underlying data in the database, it is no big deal.

**Pull request merged**